src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1074,18 +1064,51 @@ static bool saveValidatedLedger (
     }
 
     {
+        static std::string addLedger(
+            "INSERT OR REPLACE INTO Ledgers "
+            "(LedgerHash,LedgerSeq,PrevHash,TotalCoins,ClosingTime,PrevClosingTime,"
+            "CloseTimeRes,CloseFlags,AccountSetHash,TransSetHash) VALUES "
+            "(:ledgerHash,:ledgerSeq,:prevHash,:totalCoins,:closingTime,:prevClosingTime,"
+            ":closeTimeRes,:closeFlags,:accountSetHash,:transSetHash);");
+        static std::string updateVal(
+            "UPDATE Validations SET LedgerSeq = :ledgerSeq, InitialSeq = :initialSeq "
+            "WHERE LedgerHash = :ledgerHash;");
+
         auto db (app.getLedgerDB ().checkoutDb ());
 
-        // TODO(tom): ARG!
-        *db << boost::str (
-            addLedger %
-            to_string (ledger->info().hash) % seq % to_string (ledger->info().parentHash) %
-            to_string (ledger->info().drops) %
-            ledger->info().closeTime.time_since_epoch().count() %
-            ledger->info().parentCloseTime.time_since_epoch().count() %
-            ledger->info().closeTimeResolution.count() %
-            ledger->info().closeFlags % to_string (ledger->info().accountHash) %
-            to_string (ledger->info().txHash));
+        soci::transaction tr(*db);
+
+        auto const hash = to_string (ledger->info().hash);
+        auto const parentHash = to_string (ledger->info().parentHash);
+        auto const drops = to_string (ledger->info().drops);
+        auto const closeTime =
+            ledger->info().closeTime.time_since_epoch().count();
+        auto const parentCloseTime =
+            ledger->info().parentCloseTime.time_since_epoch().count();
+        auto const closeTimeResolution =
+            ledger->info().closeTimeResolution.count();
+        auto const accountHash = to_string (ledger->info().accountHash);
+        auto const txHash = to_string (ledger->info().txHash);
+
+        *db << addLedger,
+            soci::use(hash),
+            soci::use(seq),
+            soci::use(parentHash),
+            soci::use(drops),
+            soci::use(closeTime),
+            soci::use(parentCloseTime),
+            soci::use(closeTimeResolution),
+            soci::use(ledger->info().closeFlags),
```

Comment:
Nit: This works. However, we avoid function calls within `soci::use` because of the dangers of taking the addresses of temps (even though it's OK in this case). I'd prefer if function calls are banned from `soci::use`, it will save time from always having to check that `info()` returns a ref.